### PR TITLE
feat: add automation to publish container image

### PR DESCRIPTION
**Pull request by rawkode**
## Proposed changes

Publish a container image to GitHub's container registry

## Types of changes

What types of changes does your code introduce to Pizzly?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/bearer/pizzly/blob/master/README.md#contributing-guide) guide
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

NA

### CODE REVIEW

File: .github/workflows/publish-container-image.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      - master
+    tags:
+      - "*"
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
```
Comment (RadekMolenda) [inline .github/workflows/publish-container-image.yaml:15]: I'm not 100% why qemu is needed for publishing to github registry. Would you mind explaining pls?

Comment (rawkode) [reply .github/workflows/publish-container-image.yaml:15]: It's the recommended way to use Docker on GHA, as buildx (the new Docker builder) uses it in order to provide multi platform builds.

File: .github/workflows/publish-container-image.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+name: publish-container-image
+
+on:
+  push:
+    branches:
+      - master
```
Comment (cfabianski) [inline .github/workflows/publish-container-image.yaml:6]: Let's do it only for tags

File: .github/workflows/publish-container-image.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          tags: ghcr.io/${{ github.repository_owner }}/pizzly:latest
```
Comment (cfabianski) [inline .github/workflows/publish-container-image.yaml:32]: We would need to have a way to specify a tag to the version and update the `latest` tag to that one instead right?

File: Dockerfile [hunk 4]
```diff
@@ -2,24 +2,16 @@
 
 
 # Build image 
-FROM node:13.5.0-alpine3.11
+FROM node:14-alpine
```
Comment (cfabianski) [inline Dockerfile:5]: Why not the `latest`?

File: Dockerfile [hunk 5]
```diff
@@ -2,24 +2,16 @@
...
-COPY yarn.lock ./
-COPY config ./config
-COPY integrations ./integrations/
-COPY src ./src/
-COPY tests ./tests/
-COPY views ./views/
-
-RUN yarn install 
-
+COPY . .
```
Comment (cfabianski) [inline Dockerfile:10]: I would rather be more explicit about what goes into that image and keep all the COPY steps :)

**Review by cfabianski — COMMENTED** [review]
Thanks a lot for your contribution! 
I left a few comments. Let's do this 💪

File: Dockerfile (hunk 4 above)
Comment (rawkode) [reply Dockerfile:5]: Because Node 14 is the LTS release. However, I'll update it to whatever you want.

**Comment by rawkode** [comment]
@cfabianski Updated

**Review by cfabianski — APPROVED** [review]

**Comment by cfabianski** [comment]
Thank you for your contribution!

**Comment by rawkode** [comment]
It was failing because of the `COPY config`

Reverting the image back to `ghcr.io/${{ github.repository_owner }}/pizzly:${{ github.event.ref }}` should fix it

**Comment by cfabianski** [comment]
I’ve had several fixes. That one is one of them but is fixed. I had another issue with github.event.ref because it wasn’t a valid tag

**Comment by rawkode** [comment]
Yeah, I went through them all and seen. Currently, it's failing because you don't have `ghcr.io/bearer/` prefix. If that gets fixed, I'll take a look at any other error. We could also build the image on `master` and not a tag until we get it working, to spare the tagging problem

**Pull request merged**